src/valid/interface.rs

```diff
@@ -701,6 +703,21 @@ impl super::Validator {
             bg.clear();
         }
 
+        #[cfg(feature = "validate")]
+        {
+            let used_push_constants = module
+                .global_variables
+                .iter()
+                .filter(|&(_, var)| var.space == crate::AddressSpace::PushConstant)
+                .map(|(handle, _)| handle)
+                .filter(|&handle| !info[handle].is_empty());
+            // Check if there is more than one push constant, and error if so.
+            for handle in used_push_constants.skip(1) {
```

Comment:
There's a lint here that the loop never iterates. Should I add a deny for that lint here?

Reply:
Do if let Some(handle) = iter.next()

Reply:
Though honestly, it might be clearer to write the whole thing as a loop with early returns than an iterator.

Reply:
Done.

Reply:
Oh looks like it doesn't like that either, because it prefers `.nth()`. This works now, but as @teoxoy mentioned, in the future we'll want to support accumulating errors.